lib/pbench/test/unit/agent/test_config.py

```diff
@@ -1,36 +1,233 @@
-import os
 import pytest
 import shutil
 
+from configparser import NoOptionError, NoSectionError
+from pathlib import Path
+
+from pbench import PbenchConfig
 from pbench.agent import PbenchAgentConfig
 from pbench.common.exceptions import BadConfig
 
 
-def test_invalid_config(setup, monkeypatch):
-    shutil.copyfile(
-        "./lib/pbench/test/unit/agent/config/pbench-agent.invalid.cfg",
-        str(setup["cfg_dir"] / "pbench-agent.invalid.cfg"),
-    )
-    with pytest.raises(BadConfig):
-        PbenchAgentConfig(setup["cfg_dir"] / "pbench-agent.invalid.cfg")
+class MockedConfigParser:
+    """Mocked ConfigParser"""
 
+    def __init__(self):
+        self._sections = {
+            "pbench-agent": {
+                "install-dir": "/mock/install_dir",
+                "pbench_run": "/mock/pbench-run",
+                "pbench_log": "/mock/pbench.log",
+            },
+            "results": {
+                "scp_opts": "MockStrictHostKeyChecking=no",
+                "ssh_opts": "MockStrictHostKeyChecking=no",
+            },
+        }
 
-def test_valid_config():
-    config = PbenchAgentConfig(os.environ["_PBENCH_AGENT_CONFIG"])
-    assert "pbench-agent" in config.conf
-    assert "results" in config.conf
+    def __getitem__(self, key: str) -> str:
+        """check the existence of `key` in config file"""
+        if key not in self._sections:
+            raise KeyError(key)
+        return key
```

Comment:
I took reference from `ConfigParser` and tried to mimick the functionality. Althought the return as `key` can sure be replaced with `self._sections[key]`

Reply:
Looking at the implementation of `ConfigParser` (actually, of its base class, `RawConfigParser`), it looks like the need for implementing `__getitem__()` is to provide for their configuration "proxy" feature (that is, they are returning a proxy object for the section specified by the key instead of returning the section itself).

Either way, _if_ we need this function, then it needs to return something which looks and acts like a section and not the key.  So, my question is, do we need this function?  (That is, if your tests were working with it returning the wrong thing, then I bet it's never actually being called, which suggests that we don't actually need it.  🙂)